src/core/syscall_result.hh

```diff
@@ -68,6 +68,10 @@ struct syscall_result_extra : public syscall_result<int> {
     Extra extra;
     syscall_result_extra(int result, int error, Extra e) : syscall_result<int>{result, error}, extra{std::move(e)} {
     }
+    
+    std::error_code error_code() const {
+        return this->ec();
+    }
```

Comment:
Why do you need another alias for ec()? I don't know why ec() was made protected in a struct where absolutely everything else is public (this code comes from 02722823c0, six years ago), but if you just remove the "protected" it will become visible and you don't need a second name for it.